File: Backend/src/main/java/br/ind/priceByCar/priceByCar/domain/entity/Potencia/PotenciaEntity.java
package br.ind.priceByCar.priceByCar.domain.entity.Potencia;


import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "potencia")
@Data
@NoArgsConstructor
public class PotenciaEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Size(max = 100)
    @NotBlank
    private String descricao;

    private Integer cavalaria;

    public PotenciaEntity(String descricao) {
        this.descricao = descricao;
    }
}

File: frontend/lib/Domain/entities/potencia_entity.dart
class PotenciaEntity {
  int? id;
  String? descricao;
  double? valor;
  int? cavalaria;

  PotenciaEntity(this.id, this.descricao, this.valor, this.cavalaria);

  factory PotenciaEntity.fromMap(Map<String, dynamic> map) => 
    PotenciaEntity(
      map['id'], 
      map['descricao'], 
      map['valor'], 
      map['cavalaria']
    );
}
